src/supplemental/http/http_msg.c

```diff
@@ -711,10 +712,22 @@ nni_http_req_set_uri(nni_http_req *req, const char *uri)
 int
 nni_http_req_set_method(nni_http_req *req, const char *meth)
 {
-	if ((meth != NULL) && (strcmp(meth, "GET") == 0)) {
+	int string_set_status;
+
+	if ((meth != NULL) && (nni_strcasecmp(meth, "GET") == 0)) {
 		meth = NULL;
 	}
-	return (http_set_string(&req->meth, meth));
+	if ((string_set_status = http_set_string(&req->meth, meth)) == 0) {
```

Comment:
> Seems like it breaks some CI (segfault)... Goona revisit this when I got time.

Yeah in the function `nni_http_req_set_method` is a null-pointer dereference when the passed in method is `get` because then the requests method will be set to null. So I will add a check for that to the code:

```suggestion
	if ((string_set_status = http_set_string(&req->meth, meth)) == 0 && req->method != NULL) {
```

Reply:
Fix implemented in 1c2e768